scarb/src/ops/resolve.rs

```diff
@@ -306,6 +321,74 @@ fn generate_cairo_compilation_units(
         .collect::<Result<Vec<CompilationUnit>>>()
 }
 
+fn get_cfg_with_features(
+    mut cfg_set: CfgSet,
+    features_manifest: &Option<BTreeMap<String, Vec<String>>>,
+    enabled_features: &FeaturesOpts,
+) -> Result<Option<CfgSet>> {
+    if enabled_features.features.is_empty()
+        && !enabled_features.no_default_features
+        && features_manifest.is_none()
+    {
+        // It is ok to have no features in manifest
+        // only if no features are enabled and default features are not turned off.
+        return Ok(None);
+    }
+    let Some(features) = features_manifest else {
+        bail!("No features in manifest");
+    };
+    let available_features: HashSet<String> = features.keys().cloned().collect();
+    let cli_features: HashSet<String> = enabled_features.features.iter().cloned().collect();
+
+    let mut selected_features: HashSet<String> = if !enabled_features.no_default_features {
+        cli_features
+            .union(
+                &features
+                    .get("default")
+                    .map(|f| HashSet::from_iter(f.iter().cloned()))
+                    .unwrap_or_default(),
+            )
+            .cloned()
+            .collect()
+    } else {
+        cli_features
+    };
+
+    if enabled_features.all_features {
+        selected_features.extend(available_features.iter().cloned());
+    }
+
+    // Resolve features that are dependencies of selected features.
+    let mut queue = VecDeque::new();
+    queue.extend(selected_features.clone());
+
+    while let Some(key) = queue.pop_front() {
+        if let Some(neighbors) = features.get(&key) {
+            for neighbor in neighbors.iter() {
+                if !selected_features.contains(neighbor) {
+                    selected_features.insert(neighbor.clone());
+                    queue.push_back(neighbor.clone());
+                }
+            }
+        }
+    }
+
+    let not_found_features = selected_features
+        .difference(&available_features)
+        .collect_vec();
+
+    if !not_found_features.is_empty() {
+        bail!("Unknown features: {}", not_found_features.iter().join(", "));
```

Comment:
Actually, should we maybe change it to surround every `not_found_features` element in backticks separately instead of all together? Because with the suggestion above it would be ``unknown features: `x, y` ``, but I suppose ``unknown features: `x`, `y` `` is better.

Reply:
Both work for me